apps/asap-server/test/handlers/webhooks/fetch-by-code/fetch-by-code.test.ts

```diff
@@ -20,6 +23,11 @@ describe('Fetch-user-by-code handler', () => {
     algoliaClientMock,
   );
 
+  describe('Check if algolia validUntil is in correct format(seconds)', () => {
```

Comment:
I would actually mock/spy the `algoliaClient.generateSecuredApiKey` to see what it is being called with

Reply:
And we would have to mock everything :)
I have done this to do this in FP manner :)

Reply:
thats fine, but I don't think the test actually proves that the issue was fixed. I don't mind FP but we need to make sure that `generateSecuredApiKey` was called with the correct expiration Date. With your test I could change this line: https://github.com/yldio/asap-hub/blob/2a0f349e0e2a496750cd5383d3e5f92965fe7e6f/apps/asap-server/src/handlers/webhooks/fetch-by-code/fetch-by-code.ts#L34-L36

to 
```typescript 
validUntil: 123124
```
which is pretty much where we were before when the only test we had was this https://github.com/yldio/asap-hub/blob/5f1253e5f9d4b0f19651a1962e91108a27aca7ba/apps/asap-server/test/handlers/webhooks/fetch-by-code/fetch-by-code.test.ts#L142

Reply:
and to make it more clear - the way to test such bug is to write a test that fails, then fix it and make sure it goes green